scripts/validate_docstrings.py

```diff
@@ -210,101 +261,135 @@ def examples_errors(self):
-            writer.writerow([func_name,
-                             obj_type,
-                             doc.source_file_name,
-                             doc.source_file_def_line,
-                             doc.github_url,
-                             int(doc.deprecated),
-                             int(bool(doc.summary)),
-                             int(bool(doc.extended_summary)),
-                             int(doc.correct_parameters),
-                             int(bool(doc.examples)),
-                             shared_code])
+    api_items = list(get_api_items())
+    for func_name, func, section, subsection in api_items:
+        row, key = _csv_row(func_name, func, section, subsection,
+                            in_api=1, seen=seen)
+        seen[key] = func_name
+        writer.writerow(row)
+
+    api_item_names = set(list(zip(*api_items))[0])
+    for class_ in (pandas.Series, pandas.DataFrame, pandas.Panel):
+        for member in inspect.getmembers(class_):
+            func_name = 'pandas.{}.{}'.format(class_.__name__, member[0])
+            if (not member[0].startswith('_') and
+                    func_name not in api_item_names):
+                func = _load_obj(func_name)
+                row, key = _csv_row(func_name, func, section='', subsection='',
+                                    in_api=0)
+                writer.writerow(row)
 
     return 0
 
 
 def validate_one(func_name):
-    for maxsplit in range(1, func_name.count('.') + 1):
-        # TODO when py3 only replace by: module, *func_parts = ...
-        func_name_split = func_name.rsplit('.', maxsplit=maxsplit)
-        module = func_name_split[0]
-        func_parts = func_name_split[1:]
-        try:
-            func_obj = importlib.import_module(module)
-        except ImportError:
-            pass
-        else:
-            continue
-
-    if 'module' not in locals():
-        raise ImportError('No module can be imported '
-                          'from "{}"'.format(func_name))
-
-    for part in func_parts:
-        func_obj = getattr(func_obj, part)
-
+    func_obj = _load_obj(func_name)
     doc = Docstring(func_name, func_obj)
 
     sys.stderr.write(_output_header('Docstring ({})'.format(func_name)))
-    sys.stderr.write('{}\n'.format(doc.raw_doc))
+    sys.stderr.write('{}\n'.format(textwrap.dedent(doc.raw_doc)))
 
     errs = []
+    if doc.start_blank_lines != 1:
+        errs.append('Docstring text (summary) should start in the line '
+                    'immediately after the opening quotes (not in the same '
+                    'line, or leaving a blank line in between)')
+    if doc.end_blank_lines != 1:
+        errs.append('Closing quotes should be placed in the line after '
+                    'the last text in the docstring (do not close the '
+                    'quotes in the same line as the text, or leave a '
+                    'blank line between the last text and the quotes)')
```

Comment:
There are actually some cases where I see people deliberately leaving a blank line at the end of the docstring (see eg examples in http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html. Also numpydoc does it in the example: http://numpydoc.readthedocs.io/en/latest/format.html, although not sure if it is on purpose). 

So not sure we should be that strict here.

Reply:
It's your call, but personally I think that if there is no specific reason to leave one line sometimes, and to not leave it other times, I'd define a standard way to do it.

Reply:
Somehow I have a personal preference to leave a blank line at the end .. :-) 
But since the way you check it now is most occurring way in pandas, let's pick that.